src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputAction.java

```diff
@@ -62,8 +63,8 @@ private synchronized void loadExecutions() {
                 }
             }
             if (execution != null) {
-                // TODO JENKINS-37154 we would rather not block here
-                for (StepExecution se : execution.getCurrentExecutions(true).get()) {
+                // JENKINS-37154 sometimes we must block here in order to get accurate results
+                for (StepExecution se : execution.getCurrentExecutions(true).get(1, TimeUnit.MINUTES)) {
```

Comment:
🐛  - blocking the thread for a full minute is still far too much, and we'll still see issues.  Is there any reason we'd want this to be longer than ~1 second?